tests/end2end/test_hints_html.py

```diff
@@ -113,3 +113,17 @@ def test_word_hints_issue1393(quteproc, tmpdir):
         quteproc.wait_for(message='hints: *', category='hints')
         quteproc.send_cmd(':follow-hint {}'.format(hint))
         quteproc.wait_for_load_finished('data/{}'.format(target))
+
+def test_short_dict(quteproc,tmpdir):
+    dict_file = tmpdir / 'dict'
+    dict_file.write(textwrap.dedent("""
+        a
+        b
+    """))
+    quteproc.set_setting('hints', 'mode', 'word')
+    quteproc.set_setting('hints', 'dictionary', str(dict_file))
+    quteproc.open_path('data/hints/short_dict.html')
+    quteproc.send_cmd(':hint')
+    line = quteproc.wait_for(message='Not enough words in the dictionary.')
+    line.expected = True
+    quteproc.wait_for(message='hints: *', category='hints')
```

Comment:
My idea was to test whether qutebrowser really changes to letter mode and displays hints as expected. Should that be part of the test?